github-actions/docs/determine_if_we_need_to_build.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+$ref   = getenv('GITHUB_REF');
+$repo  = getenv('GITHUB_REPOSITORY');
+$token = getenv('GITHUB_TOKEN');
+
+function buildDocs() {
+    echo "TRUE";
+    exit(0);
+}
+
+function skipDocs() {
+    echo "FALSE";
+    exit(0);
+}
+
+/**
+ * @return string[]
+ */
+function executeApiCall(string $repo, string $resource, string $token): array
+{
+    curl_init('https://api.github.com/repos/' . $repo . '/' . $resource);
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($curl, CURLOPT_HTTPHEADER, [
+        'Authorization: bearer ' . $token,
+        'User-Agent: laminas/docs',
+        'Accept: application/vnd.github+json',
+    ]);
+    $response = curl_exec($curl);
+    $payload  = json_decode($response, true);
+    $payload  = array_map(static function (string $ref): string {
+        return $ref['name'];
+    }, $payload);
+    usort($payload, $compareRefNames);
+    return $payload;
+}
+
+function extractMajorMinorVersion(string $version): string
+{
+    $matches = [];
+    preg_match('/^(?P<release>\d+\.\d+)\./', $version, $matches);
+    return $matches['release'] ?? $version;
+}
+
+if ('master' === $ref) {
+    // Assume that release branches are not in use if we detect a push to master
+    buildDocs();
+    exit(0); // redundant; placed here to note that buildDocs exits
+}
+
+if (! preg_match('/^\d+\.\d+\.x$/', $ref)) {
+    // Not a release branch
+    skipDocs(); // redundant; placed here to note that skipDocs exits
+    exit(0);
+}
+
+$compareRefNames = function ($a, $b) {
+    $a = preg_replace('/\.x$/', '.0', $a);
+    $b = preg_replace('/\.x$/', '.0', $b);
+    return $a <=> $b;
+};
+
+$branches = executeApiCall($repo, 'branches', $token);
+$branches = array_filter($branches, function ($branch) {
+    return (bool) preg_match('/^\d+\.\d+\.x$/', $branch);
+});
+$mostRecent = array_pop($branches);
+if ($mostRecent === $ref) {
```

Comment:
I might be missing something here, but I am not sure if we should build docs on the latest branch. For example in laminas/automatic-releases we have 1.3.x and latest release is 1.2.4, so branch 1.3.x. is "like old develop" and it could contains some unreleased changes (new features, etc..)

It's quite confusing atm as I am not sure where we should really push the documentation fixes.
And we cannot really build on two different branches, as documentation can be different on these.

From above I can see that we build on the latest, or... previous if it is the latest release.

Reply:
> It's quite confusing atm as I am not sure where we should really push the documentation fixes.

At the moment we need no new release to update the documentation. It's done on the _default branch_ which was the branch with the name "master" so far. In the new branching model the default branch is the "next release branch".
Therefore the question is correct here: where to push the corrections for the documentation?

Reply:
@froschdesign Pull requests, by default, go to the default branch. With the new branching model, that branch is version.next, instead of version.current. With the old model, this made submitting documentation changes easy, as you could just submit against the default branch, and then we could merge immediately — the onus was on the person merging to then also remember to merge against the develop branch.

Under the new model, there are two possibilities:

- The author of the patch submits against the stable branch. It is then up to the maintainer to ensure that the patch also goes up to the default branch (as normally, merge-ups are only created during a release).

- The author of the patch submits against the default branch. It is then up to the maintainer to decide if the patch should also be merged to the current stable branch for immediate publication.

The second is easier logistically, as it ensures that the patch will be available for all versions going forwards, and it leaves the decision about whether this is a substantive change to the maintainer, versus the author of the documentation patch.